### DISCUSSION TITLE
How to access route/query parameters in filament

### DISCUSSION THREAD

**Discussion by jwhoami** [filamentphp/filament#12815] [Help]
### Package

Panel builder

### Package Version

v3

### How can we help you?

Hi,

I need to create a registration page with an invitation code like this:
http://localhost/member/register?code=3d275e20-13ed-46f4-979e-dbaf2cc9818d

How can i capture the code 3d275e20-13ed-46f4-979e-dbaf2cc9818d and
validate if this code exists before continuing with registration process.

Reading the docs, I found a hook called beforeValidate, however I cannot find a way to
access it.

I tried request()->code but it returns null.

Thanks

**Comment by leandrocfe** [comment]
Why not MustVerifyEmail?
https://laravel.com/docs/11.x/verification#model-preparation

**Reply by jwhoami** [reply]
I am using email verification but that happens after registration is successful.
However I want to check the invitation code before the user registers or gets saved to the database.
